controllers/configurationpolicy_controller.go

```diff
@@ -2556,6 +2558,19 @@ func handleSingleKey(
 			mergedValue.(map[string]interface{}), existingValue.(map[string]interface{}))
 	}
 
+	if key == "stringData" && existingObj.UnstructuredContent()["kind"] == "Secret" {
+		// override automatic conversion from stringData to data prior to evaluation
+		existingValue = existingObj.UnstructuredContent()["data"]
+		for k, val := range existingValue.(map[string]interface{}) {
+			decodedVal, err := base64.StdEncoding.DecodeString(val.(string))
+			if err != nil {
+				message := fmt.Sprintf("Error decoding secret %s: %s", val.(string), err)
```

Comment:
Will this message be written into the policy status, or into the log? If so, I think we should be careful not to leak any of the data from the secret. In this case, it's data that couldn't be decoded (somehow), but it could still be sensitive. I think I'd rather this message just give the secret's name and the key where the error occurred (the error itself might also leak sensitive information, although I'm not sure what the error string would really look like).

Again, just me being paranoid 😆